ImportBlocks/ImportBlocks.cs

```diff
@@ -19,79 +19,80 @@ public override void Configure()
-                        fs.Write(BitConverter.GetBytes(count), 0, sizeof(uint));
-                    fs.Seek(0, SeekOrigin.End);
-                    using (Snapshot snapshot = Blockchain.Singleton.GetSnapshot())
-                        for (uint i = start; i <= end; i++)
-                        {
-                            Block block = snapshot.GetBlock(i);
-                            byte[] array = block.ToArray();
-                            fs.Write(BitConverter.GetBytes(array.Length), 0, sizeof(int));
-                            fs.Write(array, 0, array.Length);
-                            Console.SetCursorPosition(0, Console.CursorTop);
-                            Console.Write($"[{i}/{end}]");
-                        }
                 }
-            }
-            else
-            {
-                start = 0;
-                uint end = Blockchain.Singleton.Height;
-                uint count = end - start + 1;
-                string path = args.Length >= 3 ? args[2] : "chain.acc";
-                using (FileStream fs = new FileStream(path, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.None))
+                else
                 {
-                    if (fs.Length > 0)
+                    if (writeStart)
                     {
-                        byte[] buffer = new byte[sizeof(uint)];
-                        fs.Read(buffer, 0, buffer.Length);
-                        start = BitConverter.ToUInt32(buffer, 0);
-                        fs.Seek(0, SeekOrigin.Begin);
+                        fs.Write(BitConverter.GetBytes(start), 0, sizeof(uint));
                     }
-                    if (start <= end)
-                        fs.Write(BitConverter.GetBytes(count), 0, sizeof(uint));
-                    fs.Seek(0, SeekOrigin.End);
-                    using (Snapshot snapshot = Blockchain.Singleton.GetSnapshot())
-                        for (uint i = start; i <= end; i++)
-                        {
-                            Block block = snapshot.GetBlock(i);
-                            byte[] array = block.ToArray();
-                            fs.Write(BitConverter.GetBytes(array.Length), 0, sizeof(int));
-                            fs.Write(array, 0, array.Length);
-                            Console.SetCursorPosition(0, Console.CursorTop);
-                            Console.Write($"[{i}/{end}]");
-                        }
+                }
+                if (start <= end)
+                    fs.Write(BitConverter.GetBytes(count), 0, sizeof(uint));
+                fs.Seek(0, SeekOrigin.End);
+                for (uint i = start; i <= end; i++)
+                {
+                    var blocks = Blockchain.Singleton.Store.GetBlocks();
+                    var transactions = Blockchain.Singleton.Store.GetTransactions();
+                    UInt256 hash = Blockchain.Singleton.GetBlockHash(i);
+                    Block block = blocks.TryGet(hash).GetBlock(transactions);
```

Comment:
There is no difference between this way and `Blockchain.Singleton.Store.GetBlock()`.

Reply:
Different.
`GetBlock()`  
https://github.com/neo-project/neo/blob/e1fdde307153879f0a7ae64bd3e11d042284896e/neo/Persistence/Helper.cs#L31
`GetBlocks()`  
https://github.com/neo-project/neo/blob/e1fdde307153879f0a7ae64bd3e11d042284896e/neo/Persistence/LevelDB/LevelDBStore.cs#L46

Reply:
I mean:

```c#
var blocks = Blockchain.Singleton.Store.GetBlocks();
var transactions = Blockchain.Singleton.Store.GetTransactions();
UInt256 hash = Blockchain.Singleton.GetBlockHash(i);
Block block = blocks.TryGet(hash).GetBlock(transactions);
```

```c#
Blockchain.Singleton.Store.GetBlock(i);
```

There is no difference between these two ways.

Reply:
This is Store.GetBlock(index)
```
public static Block GetBlock(this IPersistence persistence, uint index)
 {
       if (index == 0) return Blockchain.GenesisBlock;
       UInt256 hash = Blockchain.Singleton.GetBlockHash(index);
       if (hash == null) return null;
       return persistence.GetBlock(hash);
 }
```
It call GetBlock(hash)
```
public static Block GetBlock(this IPersistence persistence, UInt256 hash)
 {
            TrimmedBlock state = persistence.Blocks.TryGet(hash);
            if (state == null) return null;
            if (!state.IsBlock) return null;
            return state.GetBlock(persistence.Transactions);
 }
```
Which get block from `Blocks` cache and transactions from `Transactions` cache, not new cache.
We need get block from new cache every time.

Reply:
They are the same. You can test it.

Reply:
Will test.

Reply:
Oh! yes. 
`DataCache<UInt256, BlockState> IPersistence.Blocks => GetBlocks();`
You told me. Didn't get it. Anyway let me have a test.
Will make a new commit.

Reply:
🙏dalao